Update OpenAPI learn page (#9626)
ballerina-platform/ballerina-dev-website: swan-lake/integration-tools/openapi-tool.md

            [-o | --output] <output-location>
```

> ### Sanitize OpenAPI

Reviewer: Sanitize OpenAPI what? Can we be more specific?

Author: How about changing this title to the "**OpenAPI `sanitize` command usage**" or  "**OpenAPI `sanitize` CLI usage**"

Reviewer 2: Or maybe we can use `Sanitize OpenAPI contract usage` since it gives more context on what we are doing? @lnash94 @TharmiganK

Author: In addition to that, considering the various CLI commands available in OpenAPI tool, prefer to use 'command options' instead of 'usage' for the title. The term 'usage' typically implies a broader scenario in which this functionality might be employed.

For example, in a Ballerina-to-OpenAPI context, 'usage' could include references to CLI commands, compiler plugins, or built-in plugins. 

However, in this specific section, we are focusing on CLI options. we need to address this change in other CLI title as well.

> The Ballerina to OpenAPI command supports several usages in the Ballerina OpenAPI tool as follows.

Reviewer: This sentence seems to be generic. Can we relate it to the title please?

> The Ballerina to OpenAPI command supports several usages in the Ballerina OpenAPI tool as follows.

Author: @SachinAkash01 This is a CLI that accepts an OpenAPI specification as input and generates a sanitized OpenAPI specification as output, shall we change this description according to it?

| `-i \|--input`    | This specifies the path of the Ballerina service file (e.g., `my-service.bal`).                                                                               | Mandatory          |
| `--json`          | Generate the Ballerina service to OpenAPI output as JSON. The default is YAML.                                                                                                                            | Optional           |
| `-s \| --service` | This service name is used to identify the service that needs to be documented as an OpenAPI contract.                                                                                                | Optional           |
| `-o\|--output`    | Location of the generated OpenAPI contract. If this path is not specified, the output is written to the same directory from which the command is run.                                                | Optional           |

### Sanitize OpenAPI

> The command-line arguments below can be used with the command for each particular purpose as described below.

Reviewer: Still don't think is necessary. Can just say something along the lines of the following options are available.

| `licenseURL: string`        | You can use this to add the URL details regarding the full text of the license.                                                                                                                                                                                                                                                           | Optional          |
| `embed: string`             | Turns on generating OpenAPI documentation for the service for [introspection endpoint](https://ballerina.io/spec/http/#236-openapi-specification-resources) support when used with `true` in the annotation.                                                                                                                                                                                               | Optional          |

> 

Reviewer 3: We need to update the following sections as well right?
- [Usage](https://github.com/ballerina-platform/ballerina-dev-website/blob/master/swan-lake/integration-tools/openapi-tool.md#usage)
- [Command Options](https://github.com/ballerina-platform/ballerina-dev-website/blob/master/swan-lake/integration-tools/openapi-tool.md#command-options)

Author: with regarding to the `sanitize`, `flatten` subcommand or `--with-service-type` flag ?

Reviewer 3: Regarding the sub commands

Reviewer 3: Fixed via [0ffae86](https://github.com/ballerina-platform/ballerina-dev-website/pull/9626/commits/0ffae869cfe37038505105475a276d0c539f066c)

## OpenAPI Contract Modifier

### Modify the OpenAPI Contract Using Ballerina-Friendly Naming Conventions

> The `sanitize` subcommand sanitizes the OpenAPI contract file according to the best practices of Ballerina. The Ballerina-specific name extension x-ballerina-name is added to the schemas, including query names and object property names, which cannot be modified directly.

Reviewer 3: ```suggestion
The `sanitize` subcommand sanitizes the OpenAPI contract file according to the naming conventions of Ballerina. The Ballerina-specific name extension `x-ballerina-name` is added to the schemas, including query names and object property names, which cannot be modified directly.
```

```
bal openapi sanitize [-i | --input] <openapi-contract-file-path>
                     [-o | --output] <output-file-path>
                     [-n | --name] <generated-file-name>
                     [-f | --format] [json|yaml]
                     [-t | --tags] <tag-names>
                     [--operations] <operation-names>

> ```

Reviewer: ````suggestion
```

````

> The command-line arguments below can be used with the command for each particular purpose as described below.

Reviewer: ```suggestion
The command-line options are as follows.
```

| Command option            | Description                                                                                                                                                                                                               | Mandatory/Optional |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------|

> | `-i \| --input`           | The `openapi-contract-path` command option specifies the path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`).                                                                                        | Mandatory          |

Reviewer: Where is `openapi-contract-path` coming from?

Reviewer 3: Fixed via [0ffae86](https://github.com/ballerina-platform/ballerina-dev-website/pull/9626/commits/0ffae869cfe37038505105475a276d0c539f066c)

> | `-o \| --output`          | The modified openapi file is generated at the same location from which the `bal openapi sanitize` command is executed. You can point to another directory location by using the `(-o\|--output).` flag.                   | Optional           |

Reviewer: ```suggestion
| `-o \| --output`          | The modified OpenAPI file is generated at the same location from which the `bal openapi sanitize` command is executed. You can point to another directory using the `(-o\|--output)` option.                   | Optional           |
```

| `-n \| --name`            | The given name will be used to save the sanitized OpenAPI contract. The default name is `sanitized_openapi`.                                                                                                              | Optional           |

> | `-f \| --format`          | The sanitized OpenAPI contract will be saved in the given format. The format can be either JSON or YAML.The default format is same as the input file format.                                                              | Optional           |

Reviewer: ```suggestion
| `-f \| --format`          | The sanitized OpenAPI contract will be saved in the given format. The format can be either JSON or YAML. The default format is same as the input file format.                                                              | Optional           |
```

| `-t \| --tags`            | The sanitized OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |
| `--operations`            | The sanitized OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

For example,

> **Input OpenAPI contract file:**

Reviewer: ```suggestion
**Input OpenAPI contract file:**

```

```yaml
paths:
  /albums:
    get:
      tags:
        - albums
      operationId: getAlbums
      parameters:
        - name: _artists_
          in: query
          schema:
            type: array
            items:
              type: string
      responses:
        "200":
          description: Ok
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/album"
components:
  schemas:
    album:
      required:
        - _id
        - artist
        - title
      type: object
      properties:
        _id:
          type: string
        title:
          type: string
        artist:
          type: string
      additionalProperties: false

> ```
> **Output OpenAPI contract file:**
> ```yaml

Reviewer: ````suggestion
```

**Output OpenAPI contract file:**

```yaml
````

paths:
  /albums:
    get:
      tags:
        - albums
      operationId: getAlbums
      parameters:
        - name: _artists_
          in: query
          schema:
            type: array
            items:
              type: string
          x-ballerina-name: artists   -->//Ballerina name extension
      responses:
        "200":
          description: Ok
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/Album"
components:
  schemas:
    Album:
      required:
        - _id\-
        - artist
        - title
      type: object
      properties:

>         _id\-:

Reviewer 3: The input does not have this `\-` character
```suggestion
        _id:
```

          type: string
          x-ballerina-name: id
        artist:
          type: string
        title:
          type: string
      additionalProperties: false
```

> ### Modify the OpenAPI Contract by defining named objects for inline object

Reviewer: ```suggestion
### Modify the OpenAPI Contract by defining named objects for inline objects
```

> The `fallten` subcommand makes the OpenAPI contract more readable by relocating all inline embedded schemas to the components section.

Reviewer: ```suggestion
The `flatten` subcommand makes the OpenAPI contract more readable by relocating all inline embedded schemas to the components section.
```

```
bal openapi flatten [-i | --input] <openapi-contract-file-path>
                    [-o | --output] <output-file-path>
                    [-n | --name] <generated-file-name>
                    [-f | --format] [json|yaml]
                    [-t | --tags] <tag-names>
                    [--operations] <operation-names>

> ```

Reviewer: ````suggestion
```

````

> The command-line arguments below can be used with the command for each particular purpose as described below.

Reviewer: ```suggestion
The command-line options are as follows.
```

| `-o \| --output`          | The modified openapi file is generated at the same location from which the `bal openapi flatten` command is executed. You can point to another directory location by using the `(-o\|--output).` flag.                   | Optional           |
| `-n \| --name`            | The given name will be used to save the flattened OpenAPI contract. The default name is `flattened_openapi`.                                                                                                              | Optional           |
| `-f \| --format`          | The flattened OpenAPI contract will be saved in the given format. The format can be either JSON or YAML.The default format is same as the input file format.                                                              | Optional           |
| `-t \| --tags`            | The falttened OpenAPI contract will only have the operations with the given tags.                                                                                                                                         | Optional           |

> | `--operations`            | The falltened OpenAPI contract will only have the given operations.                                                                                                                                                       | Optional           |

Reviewer: Similar comments as above apply here.

Reviewer 3: Fixed via [0ffae86](https://github.com/ballerina-platform/ballerina-dev-website/pull/9626/commits/0ffae869cfe37038505105475a276d0c539f066c)

For example,

> **Input OpenAPI contract file:**

Reviewer: ```suggestion
**Input OpenAPI contract file:**

```

| `licenseURL: string`        | You can use this to add the URL details regarding the full text of the license.                                                                                                                                                                                                                                                           | Optional          |
| `embed: string`             | Turns on generating OpenAPI documentation for the service for [introspection endpoint](https://ballerina.io/spec/http/#236-openapi-specification-resources) support when used with `true` in the annotation.                                                                                                                                                                                               | Optional          |

## OpenAPI Contract Modifier

> ### Modify the OpenAPI Contract Using Ballerina-Friendly Naming Conventions

Reviewer: ```suggestion
### Modify the OpenAPI contract using Ballerina-friendly naming conventions
```

Check everywhere.

Reviewer: I think it's more preferred than friendly.
